### PR TITLE
Update default artifact name

### PR DESCRIPTION
**Pull request by jahsu-MSFT**
Update the default pipeline artifact name to an empty string and set required to false.

### CODE REVIEW

File: Tasks/AzureResourceGroupDeploymentV2/package-lock.json [hunk 1]
```diff
@@ -20,11 +20,6 @@
         "vsts-task-lib": "2.0.1-preview"
       },
       "dependencies": {
-        "node-uuid": {
```
Comment (fadnavistanmay) [inline Tasks/AzureResourceGroupDeploymentV2/package-lock.json]: Why is this file modified?

File: Tasks/PublishPipelineArtifactV0/task.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 0,
-        "Minor": 151,
+        "Minor": 139,
```
Comment (fadnavistanmay) [inline Tasks/PublishPipelineArtifactV0/task.json]: The version was 151 before?

Comment (jahsu-MSFT) [reply Tasks/PublishPipelineArtifactV0/task.json]: Yeah, since your changes to the master is not merged yet.

**Comment by jahsu-MSFT** [comment]
Will only merge after the code changes in the agent is rolled out.
